rust/sdk/examples/chat/display.rs

```diff
@@ -127,6 +83,86 @@ pub async fn display(mut cell: ActoCell<Display, impl ActoRuntime>) {
     }
 }
 
+fn render<W: io::Write>(
+    f: &mut Frame<CrosstermBackend<W>>,
+    identity: &Reader<Identity>,
+    text_area: &Reader<TextArea>,
+    messages: &Reader<Vec<Message>>,
+    not_connected: &Option<String>,
+) {
+    identity.project(|identity| match &identity.edit {
+        Some(edit) => render_editing_identity(f, edit),
+        None => render_chat(f, text_area, messages, not_connected),
+    });
+}
+
+fn render_editing_identity<W: io::Write>(f: &mut Frame<CrosstermBackend<W>>, edit: &TextArea) {
+    let size = f.size();
+    let layout = Layout::default()
+        .direction(ratatui::prelude::Direction::Vertical)
+        .constraints([ratatui::prelude::Constraint::Length(1)])
```

Comment:
does that put the edit block at the top of the terminal? should probably be in the middle

Reply:
It does. I don't know, I think text in the middle of a terminal is unnatural. But you can convince me otherwise

Reply:
`irssi` has text input only at the bottom; various Unix terminal graphics interfaces place popup boxes in the middle of the screen, e.g. Debian’s package configuration system.

But to really answer this question we’d have to actually do some UI design for this chat application.

Reply:
> various Unix terminal graphics interfaces place popup boxes in the middle of the screen

I see. I am not aware of this culture. No problem then, since this part of the design seems to be more of an interactive process